capstone/src/main/java/com/google/sps/data/Classroom.java

```diff
@@ -1,94 +1,100 @@
+import com.google.sps.service.DatabaseService;
 
 public class Classroom {
+    protected static final String CLASSROOM = "Classroom";
+    protected static final String TEACHER = "teacher";
+    protected static final String SUBJECT = "subject";
+    protected static final String STUDENTS = "students";
+    protected static final String FORMS = "forms";
 
-    private Teacher teacher;
-    private List<Student> students = new ArrayList<>();
-    private List<Form> forms = new ArrayList<>();
-    private String subject;
-    private Key key;
-
-    public Classroom(Teacher teacher, List<Student> students, List<Form> forms, String subject, Key key) {
-        this.teacher = teacher;
-        this.students = students;
-        this.forms = forms;
-        this.subject = subject;
-        this.key = key;
-    }
+    protected Entity entity;
 
-    // Setters
-    public void setTeacher(Teacher teacher) {
-        this.teacher = teacher;
+    public Classroom(Entity entity) {
+        this.entity = entity;
     }
 
-    public void setSubject(String subject) {
-        this.subject = subject;
+    public Classroom(User teacher, String subject) { 
+        this.entity = new Entity(CLASSROOM);
+        this.entity.setProperty(TEACHER, teacher.getUserKey());
+        this.entity.setProperty(SUBJECT, subject);
+        this.entity.setProperty(STUDENTS, new ArrayList<>());
+        this.entity.setProperty(FORMS, new ArrayList<>());
     }
 
-    public void setKey(Key key) {
-        this.key = key;
+    public void setTeacher(User teacher) { 
+        this.entity.setProperty(TEACHER, teacher.getUserKey());
     }
-
-    // Getters
-    public Teacher getTeacher() {
-        return this.teacher;
+    
+    public Key getTeacher() {
+        return (Key) this.entity.getProperty(TEACHER);
     }
 
     public String getSubject() {
-        return this.subject;
+        return (String) this.entity.getProperty(SUBJECT);
     }
 
-    public List<Student> getAllStudents() {
-        return this.students;
+    public Entity getClassroomEntity() {
+        return this.entity;
     }
 
-    public List<Form> getForms() {
-        return this.forms;
+    @SuppressWarnings("unchecked")
+    public List<Key> getAllStudents() {
+        return (ArrayList<Key>) this.entity.getProperty(STUDENTS);
     }
 
-    public Key getKey() {
-        assert this.key.getId() != -1: "The key doesn't exist.";
-        return this.key;
+    @SuppressWarnings("unchecked")
+    public List<Key> getAllForms() { 
+        return (ArrayList<Key>) this.entity.getProperty(FORMS); 
     }
+    
+    @SuppressWarnings("unchecked")
+    public void addStudent(User student) {
+        List<Key> students = (ArrayList<Key>) this.entity.getProperty(STUDENTS);
+        students.add(student.getUserKey());
 
-    // Database
-    public void addStudent(Student student) { 
-        students.add(student);
+        this.entity.setProperty(STUDENTS, students);
```

Comment:
Have you tried the code without this line? Im not sure which type the API exposes, but it might A: be copying in which case this is necessary, or B: return the backing list or a list subclass which affects the underlying entity when you manipulate the list